core/src/test/scala/unit/kafka/tools/DumpLogSegmentsTest.scala

```diff
@@ -154,12 +126,42 @@ class DumpLogSegmentsTest {
       new EndTransactionMarker(ControlRecordType.COMMIT, 100)
     ), origin = AppendOrigin.COORDINATOR, leaderEpoch = 7)
 
-    assertDumpLogRecordMetadata()
+    assertDumpLogRecordMetadata(log)
+
+    closeTestLog(log)
+  }
+
+  private def createTestLog = {
```

Comment:
This is to avoid creating the log twice. Each test may have a different log configuration.